### PR TITLE
Login as trial user

### PR DESCRIPTION
**Pull request by daniyarchambylov**

### CODE REVIEW

**Comment by daniyarchambylov** [comment]
@fgimenez  I didn't add trial user authentication here yet. It will be done in separate PR. For now login with userId is treated same way ethAddress is. When trial userId will be provided I will also add changes to accept empty password and Client instantiation with no params.

**Comment by daniyarchambylov** [comment]
@fgimenez  you said that we already can create  object with ` const client = new armlet.Client()` but I don't see changes that would allow it https://github.com/ConsenSys/armlet/blob/master/index.js#L14
If you have those changes which branch is it?  Or I can do it myself :)

**Comment by rocky** [comment]
@daniyarchambylov @fgimenez let's hold off on merging this until after release 1.0.0. It doesn't look like a bug fix but an improvement. And it will be coordinated with other improvements that we will need in truffle-analyze.

**Comment by daniyarchambylov** [comment]
changed base branch for PR. It shouldn't affect master branch @rocky @fgimenez

File: index.js [hunk 1]
```diff
@@ -15,13 +15,13 @@ class Client {
       throw new TypeError('Please provide auth options.')
     }
 
-    const { email, ethAddress, apiKey, password } = auth
+    const { email, ethAddress, apiKey, password, userId } = auth
```
Comment (fgimenez) [inline index.js]: Do we really need to add `userId` to the `auth` object? If the intended usage is instantiating the client without parameters, IMO we should not expect the user providing a `userId`, is that correct?

File: index.js [hunk 2]
```diff
@@ -15,13 +15,13 @@ class Client {
...
 
-    if (!email && !ethAddress && !apiKey) {
+    if (!email && !ethAddress && !apiKey && !userId) {
```
Comment (fgimenez) [inline index.js]: Same as above, I think it is ok to not pass `userId`, in that case we will later initialize it to the trial user ID.

File: index.js [hunk 3]
```diff
@@ -15,13 +15,13 @@ class Client {
...
       throw new TypeError('Please provide an user id auth option.')
     }
 
-    if (!apiKey && (!password && (email || ethAddress))) {
+    if (!apiKey && (!password && (email || ethAddress || userId))) {
```
Comment (fgimenez) [inline index.js]: Same as above, `password` should be only required when there's no `apiKey` and there's `email` or `ethAddress`. Once we know that we are dealing with the trial user (no credentials given), we should not care if the password is missing for `userId`.

**Review by fgimenez — COMMENTED** [review]
The login part looks good, not sure about the input validations given that we are going to initialize the trial user, let me know WDYT.

**Comment by daniyarchambylov** [comment]
@fgimenez  I see your point. If we'll use userIid for trial user authentication then there is no need to pass userId. I'll fix PR

File: index.js [hunk 4]
```diff
@@ -8,20 +8,22 @@ const refresh = require('./lib/refresh')
...
+      userId = trialUserId
+    }
 
-    if (!email && !ethAddress && !apiKey) {
+    if (password && !email && !ethAddress && !apiKey) {
       throw new TypeError('Please provide an user id auth option.')
     }
 
-    if (!apiKey && (!password && (email || ethAddress))) {
+    if (!apiKey && userId !== trialUserId && (!password && (email || ethAddress))) {
```
Comment (fgimenez) [inline index.js]: I think we should remove `userId !== trialUserId` from the condition, with the current code it seems not possible given that we are setting it and it is not set from outside, WDYT? Maybe check for `!userId`?

**Review by fgimenez — COMMENTED** [review]
Thanks! Only one question about a condition, other than that LGTM

**Review by fgimenez — APPROVED** [review]
Thanks!

**Pull request merged**